### DISCUSSION TITLE
Is there a way to propagate port templates from a device type to all pre-existing instances of that device?

### DISCUSSION THREAD

**Discussion by Almeus** [netbox-community/netbox#8689] [Q&A]
I added a bunch of device instances before understanding how ports/interfaces are modeled in netbox.
Reading the documentation, it says if you have a template set up on a device type, all future instances of the device will be given the ports defined in the template. I know I can add ports to a group of devices, but it won't let me select every instance at once, Even the "select all records that match the query" box doesn't work. If I check it, then go to add a component, it just grabs the instances that were on the screen at the time.

Also, this may be a case of me not reading thoroughly enough, but how can I update existing ports, as opposed add new ports, to all instances of a port across all devices of a type? Say I thought a port was a console port but turned out to be a modem port.

**Comment by candlerb** [comment]
Duplicate of #8189.  See:

https://github.com/netbox-community/reports/blob/master/reports/dcim-reports/missing_device_type_components.py (report)
https://github.com/netbox-community/reports/blob/master/scripts/add_device_type_components.py (custom script)

**Reply by Almeus** [reply]
The first script you linked to doesn't work. It's also only supposed to let you know which devices deviate from the template rather than actually applying any changes. The second script you linked to only operates on one device instance at a time. I have dozens of instances of each device.

**Reply by candlerb** [reply]
The first script is indeed just a report: it "works" as intended to highlight where there are differences.

The second script has a multi-select widget (`MultiObjectVar`), so you should be able to select multiple devices here.

If you don't like them, please feel free to edit them to meet your needs.
